### ISSUE TITLE
The client waits infinitely for the hub to deposit

### ISSUE THREAD

**Issue by kerzhner** [statechannels/monorepo#1974]
Client waits on "The hub is now depositing funds on-chain. This may take a moment." There are no logs on the hub with the address of the client. This implies that the hub never sees any messages from the client.
[web3torrent.Tue, 26 May 2020 16_21_12 GMT.5fd1c665f.0x3D23C9057B3DF0825734C73D75516a4a6Fd646E3.pino.log](https://github.com/statechannels/monorepo/files/4683578/web3torrent.Tue.26.May.2020.16_21_12.GMT.5fd1c665f.0x3D23C9057B3DF0825734C73D75516a4a6Fd646E3.pino.log)
<img width="1280" alt="Screen Shot 2020-05-26 at 9 29 31 AM" src="https://user-images.githubusercontent.com/1062379/82926027-6e3f1d80-9f33-11ea-8af2-cb4c678f9c1b.png">

**Comment by kerzhner** [comment]
Last transition is:
```
 "module": "wallet",
  "id": "APPROVE_BUDGET_AND_FUND-0x3D23C9057B3DF0825734C73D75516a4a6Fd646E3-firebase:simple-hub",
  "from": { "deposit": "init" },
  "to": { "deposit": "waitTurn" },
  "event": {
    "type": "CHAIN_EVENT",
    "balance": { "_hex": "0x00", "_isBigNumber": true },
    "blockNum": { "_hex": "0x79b1b1", "_isBigNumber": true }
  }
```

**Comment by lalexgap** [comment]
> Last transition is:
> 
> ```
>  "module": "wallet",
>   "id": "APPROVE_BUDGET_AND_FUND-0x3D23C9057B3DF0825734C73D75516a4a6Fd646E3-firebase:simple-hub",
>   "from": { "deposit": "init" },
>   "to": { "deposit": "waitTurn" },
>   "event": {
>     "type": "CHAIN_EVENT",
>     "balance": { "_hex": "0x00", "_isBigNumber": true },
>     "blockNum": { "_hex": "0x79b1b1", "_isBigNumber": true }
>   }
> ```

Do you see any logs for assetHolder[0] events?

**Comment by kerzhner** [comment]
> Do you see any logs for assetHolder[0] events?

Unfortunately, the assetHolder log line was introduced yesterday. And that is before this issue was created.

**Comment by kerzhner** [comment]
The description of the issue is likely incorrect. We see `GOT FROM FIREBASE:` in the logs. Which means that the hub has received and replied to messages from the client.

**Comment by kerzhner** [comment]
The heroku production logs from 2 days ago are no longer available. There are 2 possibilities:
1. The hub never deposited. We need hub logs to verify this.
1. The client never noticed the deposited event. Additional log lines from yesterday would help diagnose this.

In either case, we'll need to reproduce this issue and record client and hub logs to debug further.

**Comment by lalexgap** [comment]
I suspect that this is the same underlying issue as #1995 but just occurring earlier. In #1995 we see the event fire but the query return "stale" data after the user has submitted the transaction. The same thing might be happening here where the event fires from the hub deposit but the query to chain returns "stale" data causing w3t to wait indefinitely.

**Comment by lalexgap** [comment]
> I suspect that this is the same underlying issue as #1995 but just occurring earlier. In #1995 we see the event fire but the query return "stale" data after the user has submitted the transaction. The same thing might be happening here where the event fires from the hub deposit but the query to chain returns "stale" data causing w3t to wait indefinitely.

I'm starting to think this may be a separate issue (#1993/#2031). I've had this happen locally when running an integration test and I don't see any `Deposited` event in the xstate-wallet logs.

**Comment by lalexgap** [comment]
> I'm starting to think this may be a separate issue (#1993/#2031). I've had this happen locally when running an integration test and I don't see any `Deposited` event in the xstate-wallet logs.

Nevermind, it looks like this was a goof on my part. I had extended a test to use multiple leachers but due to the amount of leachers I had one of them was using the hub's private key.
